subsys/net/ip/connection_mgr.c

```diff
@@ -0,0 +1,83 @@
+/** @file
+ * @brief Generic connection manager related routines.
+ */
+
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(net_conn_mgr, CONFIG_NET_CONNECTION_MANAGER_LOG_LEVEL);
+
+#include <net/net_core.h>
+#include <net/net_ip.h>
+#include <net/net_if.h>
+#include <net/net_mgmt.h>
+#include <net/connectivity.h>
+
+/*
+ * TODO:
+ *
+ * - Create a monitor that could periodically try to check if the connectivity
+ *   is still valid. How this period should be done is the tricky part.
+ */
+
+void net_conn_mgr_connect(struct net_if *iface, sa_family_t family)
+{
+	/* We have IP adderess added to the system, so possibly we have
+	 * a proper network connection established atm.
+	 */
+	if (net_if_flag_is_set(iface, NET_IF_CONNECTED)) {
+		return;
+	}
+
+#if defined(CONFIG_NET_CONNECTIVITY_CHECK)
+	if (iface->connectivity_check) {
+		enum net_verdict;
```

Comment:
Doesnt this need to enum net_verdict <variable_name>?

Reply:
Yep, you are right. I will remove the check in next version as we need to re-think how to implement this anyway.